auacm/angular/src/app/auth.service.ts

```diff
@@ -0,0 +1,189 @@
+        resolve(false);
+      });
+    });
+  }
+
+  getUserData() : UserData {
+    return this.userData;
+  }
+  
+  me() : Promise<UserData> {
+    return new Promise((resolve, reject) => {
+      var self = this;
+      this._http.get('/api/me').subscribe((res: Response) => {
+        if (res.status == 200) {
+          var data = res.json().data;
+          var userData = new UserData();
+          userData.displayName = data.displayName;
+          userData.isAdmin = data.isAdmin === 1;
+          userData.loggedIn = true;
+          userData.username = data.username;
+          self.updateUserData(userData);
+          resolve(userData);
+        } else {
+          resolve(new UserData());
+        }
+      }, (err: Response) => {
+        resolve(new UserData());
+      });
+    });
+  };
+
+  createUser(username: string, password: string, displayName: string) : Promise<SimpleResponse> {
+    return new Promise((resolve, reject) => {
+      var params = new URLSearchParams();
+      params.append('username', username);
+      params.append('password', password);
+      params.append('display', displayName);
+
+      var headers = new Headers();
+      headers.append('Content-Type', 'application/x-www-form-urlencoded');
+
+      this._http.post('/api/create_user', params.toString(), { headers: headers }).subscribe((res: Response) => {
+        if (res.status === 200) {
+          resolve(new SimpleResponse(true, 'User created successfully.'))
+        } else {
+          resolve(new SimpleResponse(false, res.json().error));
+        }
+      }, (err: Response) => {
+        if (err.status === 401) {
+          resolve(new SimpleResponse(false, 'You need to be an admin to do this.'))
+        } else if (err.status === 400) {
+          resolve(new SimpleResponse(false, 'That user already exists.'));
+        } else {
+          resolve(new SimpleResponse(false, 'Failed to create the user.'));
+        }
+      });
+    });
+  };
+
+  changePassword(oldPassword: string, newPassword: string) : Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      var params = new URLSearchParams();
+      params.append('oldPassword', oldPassword);
+      params.append('newPassword', newPassword);
+
+      var headers = new Headers();
+      headers.append('Content-Type', 'application/x-www-form-urlencoded');
+
+      this._http.post('/api/change_password', params.toString(), { headers: headers }).subscribe((res: Response) => {
+        if (res.status === 200) {
+          resolve(true);
+        } else {
+          resolve(false);
+        }
+      }, (err: Response) => {
+        resolve(false);
+      })
+    })
+  };
+
+  /**
+   * We should implement this sometime in the future so that we don't have to edit the
+   * database manually if we want to make someone an admin.
+   */
+  updateAdminStatus(username: string) : Promise<SimpleResponse> {
+    return undefined;
+  }
+
+  getRanking(timeframe: string) : Promise<RankData[]> {
```

Comment:
If you plan on splitting up the AuthService and UserService, that, and maybe change password, sound like a good division, yes.